### PR TITLE
[RNMobile] Update gesture handler and reanimated libraries

### PR DESCRIPTION
**Pull request by fluiddot**
**Gutenberg Mobile PR:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/4615

## Description

This PR updates the following libraries:
- `react-native-gesture-handler` to version `2.2.0` ([forked repository PR](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/15))
- `react-native-reanimated` to version `2.2.4` ([forked repository PR](https://github.com/wordpress-mobile/react-native-reanimated/pull/11))

Additionally, due to API changes in the newer versions of both libraries, a set of modifications have been applied.

### Test example
1. Apply the following patch diff that incorporates an example that uses the new APIs of the libraries:
<details><summary>Click here to display the patch diff</summary>

```diff
diff --git a/packages/edit-post/src/components/visual-editor/index.native.js b/packages/edit-post/src/components/visual-editor/index.native.js
index 886e56e493938b4b63c08199e39fc90f7accf4e5..25850c6b5cab4e0f2808064fdbd79c7c2481c0d7 100644
--- a/packages/edit-post/src/components/visual-editor/index.native.js
+++ b/packages/edit-post/src/components/visual-editor/index.native.js
@@ -6,12 +6,77 @@ import { BlockList } from '@wordpress/block-editor';
 /**
  * External dependencies
  */
-import { Keyboard } from 'react-native';
+import { Keyboard, StyleSheet, View } from 'react-native';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, {
+	useSharedValue,
+	useAnimatedStyle,
+	withSpring,
+} from 'react-native-reanimated';
 /**
  * Internal dependencies
  */
 import Header from './header';
 
+const TestComponent = () => {
+	const ballStyles = StyleSheet.create( {
+		container: {
+			position: 'absolute',
+			bottom: 64,
+			right: 32,
+			zIndex: 10,
+		},
+		ball: {
+			width: 100,
+			height: 100,
+			borderRadius: 100,
+			backgroundColor: 'blue',
+		},
+	} );
+
+	const isPressed = useSharedValue( false );
+	const offset = useSharedValue( { x: 0, y: 0 } );
+	const animatedStyles = useAnimatedStyle( () => {
+		return {
+			transform: [
+				{ translateX: offset.value.x },
+				{ translateY: offset.value.y },
+				{ scale: withSpring( isPressed.value ? 1.2 : 1 ) },
+			],
+			backgroundColor: isPressed.value ? 'yellow' : 'blue',
+		};
+	} );
+
+	const start = useSharedValue( { x: 0, y: 0 } );
+	const gesture = Gesture.Pan()
+		.onBegin( () => {
+			isPressed.value = true;
+		} )
+		.onUpdate( ( e ) => {
+			offset.value = {
+				x: e.translationX + start.value.x,
+				y: e.translationY + start.value.y,
+			};
+		} )
+		.onEnd( () => {
+			start.value = {
+				x: offset.value.x,
+				y: offset.value.y,
+			};
+		} )
+		.onFinalize( () => {
+			isPressed.value = false;
+		} );
+
+	return (
+		<View style={ ballStyles.container }>
+			<GestureDetector gesture={ gesture }>
+				<Animated.View style={ [ ballStyles.ball, animatedStyles ] } />
+			</GestureDetector>
+		</View>
+	);
+};
+
 export default class VisualEditor extends Component {
 	constructor( props ) {
 		super( props );
@@ -58,11 +123,14 @@ export default class VisualEditor extends Component {
 		const { isAutoScrollEnabled } = this.state;
 
 		return (
-			<BlockList
-				header={ this.renderHeader }
-				safeAreaBottomInset={ safeAreaBottomInset }
-				autoScroll={ isAutoScrollEnabled }
-			/>
+			<>
+				<TestComponent />
+				<BlockList
+					header={ this.renderHeader }
+					safeAreaBottomInset={ safeAreaBottomInset }
+					autoScroll={ isAutoScrollEnabled }
+				/>
+			</>
 		);
 	}
 }
```

</details>

2. Open any post/page.
3. Observe that a blue ball is displayed in the right lower corner.
4. Observe that the ball can be dragged and dropped around the screen.

### Unit tests pass
1. Run `npm run native test` and observe that all tests pass.
1. Check that PR check `Unit Tests / Mobile` pass.

## Screenshots 
N/A

## Types of changes

Dependencies update

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

**Comment by fluiddot** [comment]
The unit tests are failing because the new version of `react-native-gesture-handler` shows the following warning:
```
[react-native-gesture-handler] Seems like you're using an old API with gesture components, check out new Gestures system!
```

This is produced by the React Navigation library, as it still uses the old APIs but since it's just a warning, we should proceed to ignore it. However, in the tests, we can't silence it by using `LogBox` as it's not enabled in the testing environment hence, we have to figure out another workaround.

File: packages/react-native-editor/babel.config.js [hunk 1]
```diff
@@ -19,6 +19,10 @@ module.exports = function ( api ) {
 					extensions: [ 'css', 'scss', 'sass' ],
 				},
 			],
+			path.resolve(
+				__dirname,
+				'../../node_modules/react-native-reanimated/plugin'
+			),
```
Comment (fluiddot) [inline packages/react-native-editor/babel.config.js:25]: The installation instructions from the Reanimated library ([reference](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation/#babel-plugin)) specify to use include here:
```
'react-native-reanimated/plugin',
```

However, that approach doesn't work when this configuration is referenced from a different location, as we do in GB-mobile ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/babel.config.js)). For this reason, I had to reference the plugin by resolving its path.

File: packages/react-native-editor/ios/Podfile.lock [hunk 2]
```diff
@@ -228,10 +228,10 @@ PODS:
     - React-Core
   - react-native-slider (3.0.2-wp-2):
     - React-Core
-  - react-native-video (5.0.2-wp-2):
+  - react-native-video (5.2.0-wp-2):
     - React-Core
-    - react-native-video/Video (= 5.0.2-wp-2)
-  - react-native-video/Video (5.0.2-wp-2):
+    - react-native-video/Video (= 5.2.0-wp-2)
+  - react-native-video/Video (5.2.0-wp-2):
```
Comment (fluiddot) [inline packages/react-native-editor/ios/Podfile.lock:234]: Looks like `react-native-video` was recently updated ([reference](https://github.com/WordPress/gutenberg/pull/38426)) but this file wasn't updated to reflect the change.

**Review by geriux — APPROVED** [review]
LGTM! I tested this on both iOS and Android using the demo app. 🚀

**Comment by fluiddot** [comment]
Heads up that I've bumped `react-native-gesture-handler` version to `2.2.0-wp-3` in https://github.com/WordPress/gutenberg/pull/39098/commits/fe1c723befdb87647d3583593c391b3fca0b3307.

**Pull request merged**